docs/dapps/getting-started.md

```diff
@@ -8,91 +8,531 @@ import UntranslatedPageText from "@site/src/components/UntranslatedPageText";
 ```
-npm run test
+
+You'll also need our `web3` package
+
+```sh
+npm install @alephium/web3 @alephium/web3-test
 ```
 
-or
+Create a `test` folder:
+
+```sh
+mkdir test
+```
 
+and add this minimal test: `test/token.test.ts`
+
+```typescript
+import { web3, Project, addressFromContractId } from '@alephium/web3'
+import { randomContractId, testAddress } from '@alephium/web3-test'
+import { TokenFaucet } from '../artifacts/ts'
+
+describe('unit tests', () => {
+  it('test withdraw', async () => {
+
+    // Use the correct host and port
+    web3.setCurrentNodeProvider('http://127.0.0.1:22973')
+    await Project.build()
+
+    let testContractId = randomContractId()
+    let testParams = {
+      // a random address that the test contract resides in the tests
+      address: addressFromContractId(testContractId),
+      // assets owned by the test contract before a test
+      initialAsset: { alphAmount: 10n ** 18n, tokens: [{ id: testContractId, amount: 10n }] },
+      // initial state of the test contract
+      initialFields: {
+        symbol: Buffer.from('TF', 'utf8').toString('hex'),
+        name: Buffer.from('TokenFaucet', 'utf8').toString('hex'),
+        decimals: 18n,
+        supply: 10n ** 18n,
+        balance: 10n
+      },
+      // arguments to test the target function of the test contract
+      testArgs: { amount: 1n },
+      // assets owned by the caller of the function
+      inputAssets: [{ address: testAddress, asset: { alphAmount: 10n ** 18n } }]
+    }
+
+    const testResult = await TokenFaucet.testWithdrawMethod(testParams)
+    console.log(testResult)
+  })
+})
 ```
+
+A more complex test can be found [here](https://github.com/alephium/nextjs-template/blob/main/test/token.test.ts)
+
+Without entering to much into details, `Typescript` needs some configuration to run the test, just copy the following into `tsconfig.json`
+
+```json
+{
+  "compilerOptions": {
+    "outDir": "dist",
+    "target": "es2020",
+    "esModuleInterop": true,
+    "module": "commonjs",
+    "resolveJsonModule": true
+  },
+  "exclude": ["node_modules"],
+  "include": ["src/**/*.ts", "test/**/*.ts", "scripts/**/*.ts", "alephium.config.ts", "artifacts/**/*.ts"]
+}
+```
+
+You can now run the test:
+
+```sh
 npx @alephium/cli@latest test
 ```
 
+Congratulations, you should see what result would produce a withdraw.
+
 ## Deploy your contract
 
-Next, to deploy the contract we will use Alephium CLI and a deployment script `scripts/0_deploy_faucet.ts`. You can run it using:
+Now things are getting serious, we will deploy our contract on our `devnet` :rocket:
+
+The `deploy` command will deploy everything which is inside the `scripts` folder.
 
+```sh
+mkdir scripts
 ```
+
+Let's create this file into this folder: `scripts/0_deploy_faucet.ts`
+Note that deployment scripts should prefixed with numbers (starting from 0)
+
+```typescript
+import { Deployer, DeployFunction, Network } from '@alephium/cli'
+import { Settings } from '../alephium.config'
```

Comment:
`Module '"../alephium.config"' has no exported member 'Settings'.`

Reply:
ah yes, maybe  I'll define a default `Settings` the first time I give the code of `alephium.config.ts`